File: briar-core/src/main/java/org/briarproject/briar/telegram/TelegramTdlibLoginClient.java
package org.briarproject.briar.telegram;

import org.briarproject.briar.api.telegram.TelegramAuthSession.RecoverableErrorDetail;
import org.briarproject.briar.api.telegram.TelegramAuthState;
import org.briarproject.nullsafety.NotNullByDefault;

@NotNullByDefault
interface TelegramTdlibLoginClient {

	TelegramAuthState start();

	RecoverableErrorDetail getRecoverableErrorDetail();

	TelegramAuthState submitIdentifier(String identifier);

	TelegramAuthState submitCode(String code);

	TelegramAuthState submitPassword(String password);

	TelegramAuthState close();
}

File: briar-core/src/main/kotlin/org/briarproject/briar/telegram/TelegramTdlibLoginClient.kt
package org.briarproject.briar.telegram

import org.briarproject.briar.api.telegram.TelegramAuthSession.RecoverableErrorDetail
import org.briarproject.briar.api.telegram.TelegramAuthState

interface TelegramTdlibLoginClient {
	fun start(): TelegramAuthState
	fun getRecoverableErrorDetail(): RecoverableErrorDetail
	fun submitIdentifier(identifier: String): TelegramAuthState
	fun submitCode(code: String): TelegramAuthState
	fun submitPassword(password: String): TelegramAuthState
	fun close(): TelegramAuthState
}
